### PR TITLE
Optimize ColumnNumberToName function performance, reduce about 50% memory usage and 50% time cost

### PR DESCRIPTION
**Pull request by zhayt**
# PR Details

**Optimizing memory usage by pre-allocation.**

## Description

1. **ColumnNumberToName Function**:
   - The `ColumnNumberToName` function previously concatenated rows using the `+` operator, leading to significant memory allocation when called frequently. 
   - The function now uses a pre-allocated slice to build the Excel column name. During construction, the converted number to letter is added to this slice. At the end, the slice of runes is converted to a string and returned.

2. **Sheet Pre-allocation**:
   - Added an option to create a Sheet with pre-allocated rows. 
   - The `xlsxWorksheet` structure now contains a field with a slice type. When calling the `SetCell...` function to add a row to the sheet, it in turn calls the `prepareSheetXML` function. This function adds an `xlsxRow` structure to the slice. 
   - Previously, multiple calls to `prepareSheetXML` caused the slice to grow, allocating a lot of memory and performing unnecessary copying. The new approach pre-allocates the maximum number of rows, reducing memory allocations and copying overhead.

## Related Issue

- Resolves Issue #XXX

## Motivation and Context

- When working with a lot of data, some functions create many objects and trigger garbage collection (GC). The GC then loads the processor.
- This optimization reduces memory allocations and the associated GC overhead, improving performance.

## How Has This Been Tested

- The changes have been tested in a development environment with datasets of varying sizes.
- Benchmark tests were added to measure performance improvements in memory usage and processing time.
- Existing unit tests were run to ensure no regressions were introduced.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

---

Feel free to modify any part of this to better fit your needs!

### CODE REVIEW

File: lib.go [hunk 1]
```diff
@@ -232,12 +232,18 @@ func ColumnNumberToName(num int) (string, error) {
 	if num < MinColumns || num > MaxColumns {
 		return "", ErrColumnNumber
 	}
-	var col string
+	estimatedLength := 0
```
Comment (xuri) [inline lib.go:235]: Could you provides benchmark data for this performance improvements?

File: sheet.go [hunk 2]
```diff
@@ -30,11 +30,21 @@ import (
...
+		}
+	}
+}
+
 // NewSheet provides the function to create a new sheet by given a worksheet
 // name and returns the index of the sheets in the workbook after it appended.
 // Note that when creating a new workbook, the default worksheet named
 // `Sheet1` will be created.
-func (f *File) NewSheet(sheet string) (int, error) {
+func (f *File) NewSheet(sheet string, options ...Option) (int, error) {
```
Comment (xuri) [inline sheet.go:47]: This function allows users to set cell values ​​randomly when they can't sure the total number of rows. If user know how many rows will be written and data is in ordered, they can using stream writer for get much lower cost. So I doesn't suggest changing the function signature.

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your PR. I've left some comments.

File: lib.go (hunk 1 above)
Comment (zhayt) [reply lib.go:235]: I wrote a benchmark test for the old and new version of the function, and I also made some changes (replacing a slice of runes with a slice of bytes).

[bencgcmp_result.txt](https://github.com/user-attachments/files/16086822/bencgcmp_result.txt)

File: sheet.go (hunk 2 above)
Comment (zhayt) [reply sheet.go:47]: I saw that using StreamWriter did not help the person who reported issue 876. Maybe I should use both options under the same conditions and compare the results?

**Review by xuri — CHANGES_REQUESTED** [review]
I suggest that we separate these two improvements into 2 PRs, and I will take a reconsider about the design of reallocate rows for worksheets.

**Comment by zhayt** [comment]
Sure, I will separate these two changes into different PRs. Do I need to create two new PRs, or can I keep the changes related to the ColumnNumberToName function in this PR?

Thanks for reviewing!

**Comment by xuri** [comment]
You can keep the changes related to the `ColumnNumberToName` function in this PR. Thanks.

**Comment by zhayt** [comment]
Done, I left in this PR only the changes related to the ColumnNumberToName function. Thanks.

**Review by xuri — APPROVED** [review]

**Review by xuri — APPROVED** [review]
LGTM, thanks for your contribution.

**Pull request merged**